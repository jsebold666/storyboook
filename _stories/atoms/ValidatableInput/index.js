import React from 'react';
import { text, selectV2 as select, boolean, object } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import readme from './README.md';
import ValidatableInput from '../../../components/atoms/ValidatableInput';

const component = () => (
  <ValidatableInput
    onChange={action('changed')}
    value={text('value', 'ValidatableInput content')}
  />
);

export default [readme, component];
