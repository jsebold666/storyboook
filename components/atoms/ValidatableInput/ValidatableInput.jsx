import React, { Component } from 'react';
import styled from 'styled-components';
import shared from '../../consts/shared';

const Input = styled.input`
  color: ${shared['bluey-grey']};
  background-color: ${props => (props.isDisabled ? '#f1f5f7' : 'white')};

  input:focus {
    color: ${shared['slate']};
  }
`;

export default class ValidatableInput extends Component {
  render() {
    return <Input {...this.props} />;
  }
}
