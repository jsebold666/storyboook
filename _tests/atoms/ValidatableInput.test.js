/* globals mount */
import React from 'react';
import ValidatableInput from '../../components/atoms/ValidatableInput';

describe('Expect <ValidatableInput>', () => {
  it('to render', () => {
    const props = {
      value: 'This is text',
      onChange: () => {}
    };

    const wrapper = mount(<ValidatableInput {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('to render properly with empty', () => {
    const props = {
      value: '',
      onChange: () => {}
    };

    const wrapper = mount(<ValidatableInput {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
