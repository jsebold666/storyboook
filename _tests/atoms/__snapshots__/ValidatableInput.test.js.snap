// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <ValidatableInput> to render 1`] = `
<ValidatableInput
  onChange={[Function]}
  value="This is text"
>
  <ValidatableInput__Input
    onChange={[Function]}
    value="This is text"
  >
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "ValidatableInput__Input-xydh4l-0",
            "isStatic": false,
            "lastClassName": "dDkmrl",
            "rules": Array [
              "
  color: ",
              "#acbec7",
              ";
  background-color: ",
              [Function],
              ";

  input:focus {
    color: ",
              "#4a5365",
              ";
  }
",
            ],
          },
          "displayName": "ValidatableInput__Input",
          "render": [Function],
          "styledComponentId": "ValidatableInput__Input-xydh4l-0",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onChange={[Function]}
      value="This is text"
    >
      <input
        className="ValidatableInput__Input-xydh4l-0 dDkmrl"
        onChange={[Function]}
        value="This is text"
      />
    </StyledComponent>
  </ValidatableInput__Input>
</ValidatableInput>
`;

exports[`Expect <ValidatableInput> to render properly with empty 1`] = `
<ValidatableInput
  onChange={[Function]}
  value=""
>
  <ValidatableInput__Input
    onChange={[Function]}
    value=""
  >
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "ValidatableInput__Input-xydh4l-0",
            "isStatic": false,
            "lastClassName": "dDkmrl",
            "rules": Array [
              "
  color: ",
              "#acbec7",
              ";
  background-color: ",
              [Function],
              ";

  input:focus {
    color: ",
              "#4a5365",
              ";
  }
",
            ],
          },
          "displayName": "ValidatableInput__Input",
          "render": [Function],
          "styledComponentId": "ValidatableInput__Input-xydh4l-0",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onChange={[Function]}
      value=""
    >
      <input
        className="ValidatableInput__Input-xydh4l-0 dDkmrl"
        onChange={[Function]}
        value=""
      />
    </StyledComponent>
  </ValidatableInput__Input>
</ValidatableInput>
`;
